### DISCUSSION TITLE
Permission denied on station log file.

### DISCUSSION THREAD

**Discussion by carrzkiss** [savonet/liquidsoap#3848] [General]
Hello, all.

I have run into an issue that I did not have previously.

`Apr 02 08:46:33 linuxserver liquidsoap[786]: Fatal error: exception Sys_error("/var/log/liquidsoap/101.log: Permission denied")`

The user
```
adduser liquid liquidsoap
usermod -aG sudo liquid
```
Permissions on the folder.
liquidsoap & www-data
`drwxr-xr-x   2    liquidsoap     www-data           4096 Feb  9 18:34     liquidsoap`

script used to give permissions.
`sudo chgrp -R www-data /var/log/liquidsoap/`

What else is needed for it to allow?

**Comment by gAlleb** [comment]
HI @carrzkiss. In order for Liquidsoap to write logs it should have permissions to `/var/log/liquidsoap/` directory.

You should know which user starts the Liquidsoap in `systemd file` under `User=???` and `Group=???`and grant him permissions. If you start Liquidsoap from user `liquid` in `liquid` group then you should use `sudo chown -R liquid:liquid /var/log/liquidsoap`. If user is `liquidsoap` in `liquidsoap` group then you should use `sudo chown -R liquidsoap:liquidsoap /var/log/liquidsoap`.

If it's necessary and you've messed up any directory/file permissons you can also run:

```
sudo chmod 755 /var/log/liquidsoap
sudo find /var/log/liquidsoap/ -type f -exec chmod 644 {} \;
```

**Reply by carrzkiss** [reply]
I knew it.
I looked back at my old notes and I had the liquid user with rights to that folder.
I don't know why I didn't give it rights this go around.
I think I was just following the rights for the other liquidsoap folders and forgot all about THAT particular folder.

Thank you, Stehan.
You have an awesome day.
